subsys/storage/flash_map/Kconfig

```diff
@@ -18,8 +18,47 @@ if FLASH_MAP
 config FLASH_MAP_SHELL
 	bool "Enable flash map shell interface"
 	depends on SHELL
+	select FLASH_MAP_DEV_NAME_IN_FLASH_AREA
 	help
 	  This enables shell commands to list and test flash maps.
+	  The option depends on FLASH_MAP_DEV_NAME_IN_FLASH_AREA being selected, even when
+	  FLASH_MAP_DEV_IN_FLASH_AREA is enabled, to be able to list names of devices tied to
+	  flash areas, as the names need to be accessible from flash_area structure.
+
+config FLASH_MAP_DEV_IN_FLASH_AREA
+	bool "Switch flash map to use compile time evaluated devices objects"
+	default y if !FLASH_MAP_CUSTOM
+	help
+	  The option adds the fa_dev field to the flash_area structure that allows to evaluate flash
+	  device object pointer at compile time, and store it within flash_area structure to avoid
+	  run-time calls to obtain it.
+	  Enabling the option switches flash map API to use the flash_area.fa_dev instead of using
+	  flash_area.fa_dev_name to get device by name.
+	  In case of the default flash map, enabling this option will cause the fa_dev, of
+	  each map entry, to be populated with a pointer to the device tied to that flash area.
+	  The option is selected for default flash map (FLASH_MAP_CUSTOM=n); if your custom map
+	  populates the flash_area.fa_dev, with pointers to devices that are tied to proper flash
+	  areas, you may want to unselect this option.
```

Comment:
I don't understand this.

> The option is selected

Nothing I can see `select`s this option if FLASH_MAP_CUSTOM=n, it's just `default y` in that case. That is not the same thing.

> if your custom map [...] you may want to unselect this option.

I don't understand what this is trying to say, either.

First, I think using the word "unselect" in this context is unclear because Kconfig has a `select` keyword that you cannot undo.

Second, it seems to me that the only way to both have a custom map (`CONFIG_FLASH_MAP_CUSTOM=y`) and populate `fa_dev` pointers is if the user manually sets them up, so why would they want to make sure `CONFIG_FLASH_MAP_DEV_IN_FLASH_AREA=n`?

Reply:
I will fix the comments.

Reply:
Seems that I have merged comments from two different options.